### PR TITLE
storage: etcd: pass contexts correctly in TestConsistentList

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/kind cleanup
/sig api-machinery
/cc @stts @liggitt 

```release-note
NONE
```

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go [hunk 1]
```diff
@@ -2236,18 +2237,20 @@ func (t *fancyTransformer) createObject() error {
 		},
 	}
 	out := &example.Pod{}
-	return t.store.Create(context.TODO(), key, obj, out, 0)
+	return t.store.Create(t.ctx, key, obj, out, 0)
```
Comment (liggitt) [inline staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go:2240]: this should be plumbed as a parameter to createObject (and should use the parameter already passed to TransformFromStorage, for example)

Comment (liggitt) [reply staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go:2240]: from context godoc:

> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx

**Comment by liggitt** [comment]
/hold

for the comment about context plumbing as params instead of in the test struct

File: staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go (hunk 1 above)
Comment (stevekuznetsov) [reply staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go:2240]: @liggitt I considered trying to do that, but I'm not sure it's possible - this is called as a side-effect of transforming to storage, and there is no request context present in that call. The `context.Context` there is used for figuring out encryption.

Comment (liggitt) [reply staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go:2240]: oh, that's funny... different Context interface

it's not clear why a background context in the test struct is more correct than a TODO context passed directly to the storage create call

Comment (stevekuznetsov) [reply staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go:2240]: It's "more correct" in that one, coherent context is used in all calls to the `storage.Interface` during the whole test. In this file there's quite a lot of copy-pasta in general, but this looks to be one of the few tests that did not get changed to use `testSetup()` which divulges (a background) context that tests use to pass to all calls. For no reason other than making this not the odd-one-out, I think it's worth it.

**Comment by stevekuznetsov** [comment]
Will refactor after https://github.com/kubernetes/kubernetes/pull/108174 merges

**Pull request merged**